### PR TITLE
[breaking] Stop running in the next loop

### LINKED ISSUE #203: check interaction with render modifiers
// background - https://github.com/emberjs/ember.js/pull/18698#issuecomment-579991774
https://github.com/lifeart/ember-ref-modifier/blob/master/tests/dummy/app/components/glimmer-ref.js

since we setting property in next runloop, such code may fail
```hbs
 <div {{ref this "menuElement"}} {{did-insert this.focusOnFirstItem}} >
```
in other side, callback may solve issue

```hbs
<div {{ref this.setMenuElement}} >
<div {{did-insert this.setMenuElement}} >
```

```js
@action
setMenuElement(node) {
  this.menuElement  = node;
  this.focusOnFirstItem(node);
}
```

in other way, we can make `next runloop` sheduling optional, or remove it at all from modifier itself and give control to developers.

```hbs
{{ref this "prop"}} - for classic usage
{{ref this "prop" run="next"}}  - for runloop control
{{ref this.setElement run="next"}} - runloop controlled callback
{{ref (next-runloop this.setElement)}} - manual callback control
```

other way - allow tracked chain computations rerun in same transaction if some of tracked properties is DOM node, but it seems low-level change.

### PR DESCRIPTION
**Pull request by jrjohnson**
This moves back to setting the property synchronously and adds
instructions for handling errors which are caused by this. In this case
causing the error seems better than creating difficult to see timing
bugs when Run.next() is invoked.

Lots of context and discussion and fixes #203.

Context for the original change (reverted here) in #195

### CODE REVIEW

**Comment by jrjohnson** [comment]
@lifeart I thought I would try and save you some time by taking a pass at this as a first draft. Let me know if I'm miss-interpreting the conversation over in #203 and I can revise.

**Comment by lifeart** [comment]
@jrjohnson thank you for PR! I was thinking, https://github.com/glimmerjs/glimmer-vm/pull/1061 may  resolve original issue.

I was thinking about @tracked, we can resolve property descriptor, and if descriptor has "set" method, run action in next runloop, if not - in current

**Comment by jrjohnson** [comment]
OK, shall I close this an await a glimmer solution? We can work around this in the interim by setting the values we need using `{{did-insert}}` manually.

**Comment by lifeart** [comment]
@jrjohnson what do you think about sync/async logic, based on property descriptor?
tldr: if property defined and has getter/setter - use async implementation, if property not defined or has "simple" descriptor - use sync implementation

**Comment by jrjohnson** [comment]
Sorry @lifeart, I don't have enough understanding of the underlying tech to comment either way, I'm happy to test it out though!

**Comment by lifeart** [comment]
@jrjohnson see https://ember-twiddle.com/aa44c20fcbc9d8b318cd45d3b8d9252d?openFiles=templates.components.my-component%5C.hbs%2Ctemplates.components.my-component%5C.hbs

**Comment by lifeart** [comment]
Since this addon counted as legacy, closing this merge request in favour of https://github.com/lifeart/ember-ref-bucket
@jrjohnson thank you for your work on it and pushing this case forward! Hope updated implementation bring more clarity and stability.
